ffi/capi/src/wasm_glue.rs

```diff
@@ -61,3 +61,45 @@ pub unsafe extern "C" fn icu4x_init() {
         .map(|()| log::set_max_level(LevelFilter::Debug))
         .unwrap();
 }
+
+#[cfg(feature = "wasm_embedding")]
+mod allocator {
+    use alloc::alloc::GlobalAlloc;
+    use alloc::alloc::Layout;
+
+    // The default global allocator for Rust works fine when Rust
+    // is controlling the entire WASM stack, however when linked to
+    // C++ it does not, so we fall back to malloc/free
+    #[global_allocator]
+    static GLOBAL: WasmMalloc = WasmMalloc;
+
+    pub struct WasmMalloc;
+
+    extern "C" {
+        fn malloc(size: usize) -> *mut u8;
+        fn free(ptr: *mut u8);
+        fn calloc(num: usize, size: usize) -> *mut u8;
+        fn realloc(ptr: *mut u8, size: usize) -> *mut u8;
+    }
+    unsafe impl GlobalAlloc for WasmMalloc {
```

Comment:
Question: why not dlmalloc?

Reply:
It doesn't work! I didn't bother to debug it.